### DISCUSSION TITLE
createWeb3Modal do not detect my browser wallets => show Not Detected

### DISCUSSION THREAD

**Discussion by sylvain-guehria** [WalletConnect/walletconnect-monorepo#3885] [Developer Support] [answered]
My metamask is connected with my browser. 

When I open a web3modal, Metamask is detected in the category "Browser wallet" but if I click directly on metamask in the list (or any other wallet) it shows not detected (despite I have the wallet in my browser)

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/51234823/7b23368c-bcd2-42d6-b8dd-60873c66180c)

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/51234823/cff103ee-b487-4f95-b997-208d2d346db1)

**Comment by glitch-txs** [comment, marked as answer]
Which browser are you using? If Firefox this should be fixed. MetaMask will ship a fix by the end of the next week for other browsers.

**Reply by sylvain-guehria** [reply]
I solved the problem. It was a stupide mistake from my side. I forgote to add new EIP6963Connector({ chains }) to my connectors. 
It is the one responsible for dealing with multiples injected wallets

@glitch-txs  see my answer above, maybe it will help =)

**Reply by luismmdev** [reply]
Hey! Thanks a lot for pointing out to that connector, read about it in the release notes but didn't see it in the docs. It works flawlessly, thanks!

**Comment by luismmdev** [comment]
Hi @glitch-txs !

I'm still facing this very same issue as of today. This is not only happening with metamask, but with other wallets too (OKX Wallet, for example). I've seen the same behaviour both in chrome and firefox. Neither browser seem to be detecting the installed extension correctly when using the browser option.

**Reply by glitch-txs** [reply]
hi, try upgrading to the latest version, also could you share your code configuration?

**Reply by luismmdev** [reply]
Thanks for you tips, I've used the connector suggested by @sylvain-guehria and it works as expected. I didn't see it in the docs, sorry and thanks for your help!
